src/Microsoft.Data/BaseColumn.cs

```diff
@@ -69,6 +84,10 @@ protected set
 
         public virtual BaseColumn Sort(bool ascending = true) => throw new NotImplementedException();
 
+        public virtual Dictionary<TKey, ICollection<long>> GroupColumnValues<TKey>() => throw new NotImplementedException();
```

Comment:
You probably don't need this public method anymore, now that `GroupBy` is a public method on column.

Reply:
Right, I thought of that and decided to keep it around for the `Merge` PR. I'm ok deleting the API on BaseColumn for now and revisiting later